### PR TITLE
add codeql config

### PR DESCRIPTION
**Pull request by stan-dot**

### CODE REVIEW

**Comment by stan-dot** [comment]
```Collecting PyYAML (from blueapi==0.4.5a3.dev2+gb677a55b)
  Downloading PyYAML-5.4.1.tar.gz (175 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [48 lines of output]
      running egg_info
      writing lib3/PyYAML.egg-info/PKG-INFO```
      
      that's the weirdest error, I re-ran the script and it persisted
      
@callumforrester any tips how to solve this?

**Comment by stan-dot** [comment]
fwiw the main branch suffers from the same issue
https://github.com/DiamondLightSource/blueapi/actions/runs/10580495453/job/29315513545

**Comment by stan-dot** [comment]
investigating the cython issue
- https://github.com/yaml/pyyaml/issues/724
- https://stackoverflow.com/questions/77490435/attributeerror-cython-sources
- https://github.com/yaml/pyyaml/issues/601

seems repeating specifically in pyyaml installation

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      # 📚 See https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun
+      - if: matrix.build-mode == 'manual'
+        shell: bash
+        run: |
+          echo 'If you are using a "manual" build mode for one or more of the' \
+            'languages you are analyzing, replace this with the commands to build' \
+            'your code, for example:'
+          echo '  make bootstrap'
+          echo '  make release'
+          exit 1
```
Comment (callumforrester) [inline .github/workflows/codeql.yml:88]: Is this part needed for a Python project?

Comment (stan-dot) [reply .github/workflows/codeql.yml:88]: that was the default to be copy-pasted, I just see it as docstring

**Review by callumforrester — APPROVED** [review]
LGTM

**Pull request merged**